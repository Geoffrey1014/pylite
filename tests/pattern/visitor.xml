lambda
  name __init__
  param_count 0
  local_count 0
  label_count 0
  insn_vector
    new_class c0
    store_global c0
    load_const c1
    load_global c0
    store_field c2
    load_const c3
    load_global c0
    store_field c4
    new_class c5
    store_global c5
    load_const c6
    load_global c5
    store_field c4
    new_class c7
    store_global c7
    load_const c8
    load_global c7
    store_field c4
    new_class c9
    store_global c9
    load_const c10
    load_global c9
    store_field c2
    load_const c11
    load_global c9
    store_field c4
    new_class c12
    store_global c12
    load_const c13
    load_global c12
    store_field c14
    load_const c15
    load_global c12
    store_field c16
    load_const c17
    load_global c12
    store_field c18
    load_const c19
    load_global c12
    store_field c20
    load_global c9
    call 0
    store_global c21
    load_global c12
    call 0
    store_global c22
    load_global c22
    load_global c21
    load_field c4
    call 1
    pop
    load_const c23
    return
  const_vector
    Wheel
    lambda
      name __init__
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_local 0
        store_field c0
        load_const c1
        return
      const_vector
        name
        None
      guard_vector
      line_number_map 3 3 3 2 2
    __init__
    lambda
      name accept
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_local 1
        load_field c0
        call 1
        pop
        load_const c1
        return
      const_vector
        visitWheel
        None
      guard_vector
      line_number_map 5 5 5 5 5 4 4
    accept
    Engine
    lambda
      name accept
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_local 1
        load_field c0
        call 1
        pop
        load_const c1
        return
      const_vector
        visitEngine
        None
      guard_vector
      line_number_map 9 9 9 9 9 8 8
    Body
    lambda
      name accept
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_local 1
        load_field c0
        call 1
        pop
        load_const c1
        return
      const_vector
        visitBody
        None
      guard_vector
      line_number_map 13 13 13 13 13 12 12
    Car
    lambda
      name __init__
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        load_global c0
        call 0
        load_local 0
        store_field c1
        load_global c2
        call 0
        load_local 0
        store_field c3
        new_list 2
        load_const c4
        load_global c5
        call 1
        load_const c6
        dup_top 2
        store_item
        load_const c7
        load_global c5
        call 1
        load_const c8
        dup_top 2
        store_item
        load_local 0
        store_field c9
        load_const c10
        return
      const_vector
        Engine
        engine
        Body
        body
        "front"
        Wheel
        0.000000
        "back"
        1.000000
        wheels
        None
      guard_vector
      line_number_map 17 17 17 17 18 18 18 18 19 19 19 19 19 19 19 19 19 19 19 19 19 19 19 16 16
    lambda
      name accept
      param_count 2
      local_count 2
      label_count 3
      insn_vector
        load_local 0
        load_local 1
        load_field c0
        call 1
        pop
        load_local 1
        load_local 0
        load_field c1
        load_field c2
        call 1
        pop
        load_local 1
        load_local 0
        load_field c3
        load_field c2
        call 1
        pop
        load_local 0
        load_field c4
        iterator
        store_local -2
        jmp L1
        label L0
        store_local -1
        load_local 1
        load_local -1
        load_field c2
        call 1
        pop
        label L1
        load_local -2
        next L0
        pop
        label L2
        load_const c5
        return
      const_vector
        visitCar
        engine
        accept
        body
        wheels
        None
      guard_vector
      line_number_map 22 22 22 22 22 23 23 23 23 23 23 24 24 24 24 24 24 25 25 25 25 25 25 25 26 26 26 26 26 25 25 25 25 25 21 21
    PrintVisitor
    lambda
      name visitEngine
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_const c0
        print
        load_const c1
        return
      const_vector
        "Visiting engine"
        None
      guard_vector
      line_number_map 30 30 29 29
    visitEngine
    lambda
      name visitBody
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_const c0
        print
        load_const c1
        return
      const_vector
        "Visiting body"
        None
      guard_vector
      line_number_map 33 33 32 32
    visitBody
    lambda
      name visitWheel
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_const c0
        load_local 1
        load_field c1
        load_const c2
        add
        add
        print
        load_const c3
        return
      const_vector
        " wheel"
        name
        "Visiting "
        None
      guard_vector
      line_number_map 36 36 36 36 36 36 36 35 35
    visitWheel
    lambda
      name visitCar
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_const c0
        print
        load_const c1
        return
      const_vector
        "Visiting car"
        None
      guard_vector
      line_number_map 39 39 38 38
    visitCar
    car
    visitor
    None
  guard_vector
  line_number_map 1 1 2 2 2 4 4 4 7 7 8 8 8 11 11 12 12 12 15 15 16 16 16 21 21 21 28 28 29 29 29 32 32 32 35 35 35 38 38 38 41 41 41 42 42 42 43 43 43 43 43 1 1
