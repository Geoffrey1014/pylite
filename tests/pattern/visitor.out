Visiting car
Visiting engine
Visiting body
Visiting front wheel
Visiting back wheel
